compiler/rustc_middle/src/ty/query.rs

```diff
@@ -222,9 +222,8 @@ macro_rules! define_callbacks {
 
                 let cached = try_get_cached(self.tcx, &self.tcx.query_caches.$name, &key, noop);
 
-                let lookup = match cached {
-                    Ok(()) => return,
-                    Err(lookup) => lookup,
+                let Err(lookup) = cached else {
+                    return;
```

Comment:
The match with `Ok(())` was done on purpose here: symmetry with `TyCtxtAt::$name` below, and to emphasize that the `Ok` holds a `()`.